src/goto-harness/recursive_initialization.cpp

```diff
@@ -1008,6 +1008,21 @@ code_blockt recursive_initializationt::build_function_pointer_constructor(
   for(const auto &target : targets)
   {
     auto const assign = code_assignt{dereference_exprt{result}, target};
+    auto sym_to_lookup =
```

Comment:
:pick: `const`
:bulb: Invariant on the first operand being a symbol_exprt with specific justification for why should be a symbol

Reply:
I played around with introducing an INVARIANT, but to be honest it was very awkward - because it's a ternary operator, the whole thing needs to be expression, and for that to happen, I need to introduce a lambda if I want to introduce more than one statements followed by the expression.

I also feel that the INVARIANT is being covered by the `PRECONDITION` inside `to_symbol_expr`, but I can definitely add the justification for it.